com.unity.netcode.gameobjects/Editor/Configuration/NetcodeForGameObjectsSettings.cs

```diff
@@ -35,5 +43,39 @@ internal static void SetAutoAddNetworkObjectSetting(bool autoAddSetting)
         {
             EditorPrefs.SetBool(AutoAddNetworkObjectIfNoneExists, autoAddSetting);
         }
+
+        public static NetcodeForGameObjectsSettings GetOrCreateSettings()
```

Comment:
**Comment:** It seems a little odd to me that this class is a mixture of editor prefs and persistent prefs.
**Suggestion:** If you want to stay consistent with Unity's asset serialization and also be able to take advantage of things like `[FormerlySerializedAs]` you can use these APIs to read/write scriptable object assets to the project folder:

```
UnityEditorInternal.InternalEditorUtility.SaveToSerializedFileAndForget();
UnityEditorInternal.InternalEditorUtility.LoadSerializedFileAndForget();
```

Or alternatively, you can use `ScriptableSingleton` which does most of that for you automatically.

Reply:
I think I see what you're saying... but this doesn't actually need to be persistent prefs, it can use EditorPrefs everywhere. The mismatch was the result of merging two different implementations of this together and not noticing that they were implemented in different ways.

Reply:
I think they actually need to be separate - the existing editorprefs are really editor preferences. how should this editor on this PC behave. The prefab prefs are project-level... you want that behavior consistent on everyone's PC who is working on the project.

Reply:
Ah, fair. I didn't realize EditorPrefs were shared between projects.

Reply:
I had the same knee jerk reaction when looking at the original implementation, but there are advantages to having the NetworkPrefabs configuration serialized in reasonably "readable" json formatted text file.  This could be easily converted over to a editor preferences via [EditorPrefs.SetString](https://docs.unity3d.com/ScriptReference/EditorPrefs.SetString.html), but I am not sure a netcode project's NetworkPrefab settings (specific to the game/product) qualifies as an editor preference.

Might be worth investigating the ScriptableObject path... just to determine if we are gaining or losing potential future functionality (either side of the fence).

The good news is that the json text can be easily saved to and read from whichever medium/location turns out to be the best place to store this information.  👍

Reply:
Switched it over to ScriptableSingleton. That eased some of the logic and it's still a plain text format. :)